go-controller/pkg/node/ovspinning/ovspinning_linux.go

```diff
@@ -0,0 +1,185 @@
+//go:build linux
+// +build linux
+
+package ovspinning
+
+import (
+	"fmt"
+	"os"
+	"strconv"
+	"time"
+
+	"github.com/fsnotify/fsnotify"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/util"
+	"golang.org/x/sys/unix"
+	"k8s.io/klog/v2"
+)
+
+// These variables are meant to be used in unit tests
+var tickDuration time.Duration = 10 * time.Second
+var getOvsVSwitchdPIDFn func() (string, error) = util.GetOvsVSwitchdPID
+var getOvsDBServerPIDFn func() (string, error) = util.GetOvsDBServerPID
+var featureEnablerFile string = "/etc/openvswitch/enable_dynamic_cpu_affinity"
+
+// Run monitors OVS daemon's processes (ovs-vswitchd and ovsdb-server) and sets their CPU affinity
+// masks to that of the current process.
+// This feature is enabled by the presence of a non-empty file in the path `/etc/openvswitch/enable_dynamic_cpu_affinity`
+func Run(stopCh <-chan struct{}) {
+
+	// The file must be present at startup to enable the feature
+	isFeatureEnabled, err := isFileNotEmpty(featureEnablerFile)
+	if err != nil {
+		klog.Warningf("Can't start OVS CPU affinity pinning: %v", err)
+		return
+	}
+
+	if !isFeatureEnabled {
+		klog.Info("OVS CPU affinity pinning disabled")
+		return
+	}
+
+	fileWatcher, err := createFileWatcherFor(featureEnablerFile)
+	if err != nil {
+		klog.Warningf("Can't start OVS CPU affinity pinning: %v", err)
+		return
+	}
+	defer fileWatcher.Close()
+
+	klog.Infof("Starting OVS daemon CPU pinning")
+	defer klog.Infof("Stopping OVS daemon CPU pinning")
+
+	ticker := time.NewTicker(tickDuration)
+	defer ticker.Stop()
+
+	for {
+		select {
+		case event, ok := <-fileWatcher.Events:
+			if !ok {
+				continue
```

Comment:
wouldn't this be a failure case where the channel is closed and then someone would need to restart ovnk-node to make this work again? similar to line 62-63? What is the point of continuing here?

Reply:
This channel gets closed when `fileWatcher.Close()` is invoked, on `Run()` function exit. I think there is no chance of getting here, but I can change it to a `return` with a log if it looks cleaner.